srml/transaction-payment/src/lib.rs

```diff
@@ -0,0 +1,456 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! # Transaction Payment Module
+//!
+//! This module provides the basic logic needed to pay the absolute minimum amount needed for a
+//! transaction to be included. This includes:
+//!   - _weight fee_: A fee proportional to amount of Weight a transaction consumes. (TODO: explain
+//!     what weight is or link somewhere)
+//!   - _length fee_: A fee proportional to the encoded length of the transaction.
+//!   - _tip_: An optional tip. Tip increases the priority of the transaction, giving it a higher
+//!     chance to be included by the transaction queue.
+//!
+//! Additionally, this module allows one to configure:
+//!   - The mapping between one unit of weight to one unit of fee via [`WeightToFee`].
+//!   - A means of updating the fee for the next block, via defining a multiplier, based on the
+//!     final state of the chain at the end of the previous block. This can be configured via
+//!     [`FeeMultiplierUpdate`]
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use rstd::prelude::*;
+use codec::{Encode, Decode};
+use support::{
+	decl_storage, decl_module,
+	traits::{Currency, Get, OnUnbalanced, ExistenceRequirement, WithdrawReason},
+};
+use sr_primitives::{
+	Fixed64,
+	transaction_validity::{
+		TransactionPriority, ValidTransaction, InvalidTransaction, TransactionValidityError,
+		TransactionValidity,
+	},
+	traits::{Zero, Saturating, SignedExtension, SaturatedConversion, Convert},
+	weights::{Weight, DispatchInfo},
+};
+
+type Multiplier = Fixed64;
+type BalanceOf<T> =
+	<<T as Trait>::Currency as Currency<<T as system::Trait>::AccountId>>::Balance;
+type NegativeImbalanceOf<T> =
+	<<T as Trait>::Currency as Currency<<T as system::Trait>::AccountId>>::NegativeImbalance;
+
+pub trait Trait: system::Trait {
+	/// The currency type of the chain.
```

Comment:
actually one chain could use different currency type. 

I can suggest this,
```suggestion
	/// The currency type used.
```
but it is not better :-/

Reply:
The currency in which fees will be paid.